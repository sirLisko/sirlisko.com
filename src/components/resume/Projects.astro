---
import type { Resume } from "../../types";

interface Props {
  projects: Resume["projects"];
}
const { projects } = Astro.props;
---

<article class="section projects">
  <h1>Side Projects</h1>
  {
    projects.map((prj) => (
      <div>
        <h2>
          <a href={prj.url}>{prj.name}</a>
        </h2>
        {prj.blurp.map((blr) => (
          <p>{blr}</p>
        ))}
      </div>
    ))
  }
  <div>
    To see a full list of my side projects go to{" "}
    <a href="/projects">https://sirlisko.com/projects</a>.
  </div>
</article>
