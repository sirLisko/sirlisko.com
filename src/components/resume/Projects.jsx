import React from "react";

const Projects = ({ projects }) => (
  <article className="section projects">
    <h1>Side Projects</h1>
    {projects.map(prj => (
      <div key={prj.name}>
        <h2>
          <a href={prj.url}>{prj.name}</a>
        </h2>
        {prj.blurp.map((blr, i) => (
          <p key={i}>{blr}</p>
        ))}
      </div>
    ))}
    <div>
      To see a full list of my side projects go to{" "}
      <a href="/projects">https://sirlisko.com/projects</a>.
    </div>
  </article>
);

export default Projects;
