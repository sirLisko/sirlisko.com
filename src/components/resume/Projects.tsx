import React from "react";

interface ProjectProps {
  projects: Array<{
    name: string;
    url?: string;
    blurp: string[];
  }>;
}

const Projects = ({ projects }: ProjectProps) => (
  <article className="section projects">
    <h1>Side Projects</h1>
    {projects.map((prj) => (
      <div key={prj.name}>
        <h2>
          <a href={prj.url}>{prj.name}</a>
        </h2>
        {prj.blurp.map((blr, i) => (
          <p key={i}>{blr}</p>
        ))}
      </div>
    ))}
    <div>
      To see a full list of my side projects go to{" "}
      <a href="/projects">https://sirlisko.com/projects</a>.
    </div>
  </article>
);

export default Projects;
