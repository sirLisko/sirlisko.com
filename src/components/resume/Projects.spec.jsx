import React from "react";
import renderer from "react-test-renderer";

import Projects from "./Projects.jsx";

describe("Projects Component", () => {
  let wrapper;
  const props = {
    projects: [
      {
        name: "foo",
        blurp: ["bar", "foobar."],
        logo: "imgs/foobar.png",
        links: {
          website: "https://foobar.me",
          github: "https://github.com/sirlisko/foobar",
        },
        tech: ["foo", "bar", "foobar"],
      },
      {
        name: "asd",
        blurp: ["asd", "qwe."],
        logo: "imgs/asd.png",
        links: {
          website: "https://asd.me",
          github: "https://github.com/sirlisko/asd",
        },
        tech: ["asd", "qwe", "foobar"],
      },
    ],
  };

  beforeEach(() => {
    wrapper = renderer.create(<Projects {...props} />).toJSON();
  });

  it("should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
