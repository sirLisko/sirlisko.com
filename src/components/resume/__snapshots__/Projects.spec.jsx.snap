// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects Component should render properly 1`] = `
<article
  className="section projects"
>
  <h1>
    Side Projects
  </h1>
  <div>
    <h2>
      <a>
        foo
      </a>
    </h2>
    <p>
      bar
    </p>
    <p>
      foobar.
    </p>
  </div>
  <div>
    <h2>
      <a>
        asd
      </a>
    </h2>
    <p>
      asd
    </p>
    <p>
      qwe.
    </p>
  </div>
  <div>
    To see a full list of my side projects go to
     
    <a
      href="/projects"
    >
      https://sirlisko.com/projects
    </a>
    .
  </div>
</article>
`;
